src/System.Security.Cryptography.Xml/src/System/Security/Cryptography/Xml/KeyInfoX509Data.cs

```diff
@@ -149,11 +155,10 @@ public ArrayList IssuerSerials
 
         public void AddIssuerSerial(string issuerName, string serialNumber)
         {
-            BigInt h = new BigInt();
-            h.FromHexadecimal(serialNumber);
+            BigInteger h = BigInteger.Parse(serialNumber, NumberStyles.AllowHexSpecifier);
             if (_issuerSerials == null)
                 _issuerSerials = new ArrayList();
-            _issuerSerials.Add(new X509IssuerSerial(issuerName, h.ToDecimal()));
+            _issuerSerials.Add(new X509IssuerSerial(issuerName, h.ToString()));
```

Comment:
`X509IssuerSerial` takes a string as the second parameter, BTW. So even if I cast to `decimal` first, I'd still have to call `ToString`. The old `BigInt.ToDecimal` returned a string (which wasn't clear from the original code).